src/main/java/io/pinecone/PineconeIndexOperationClient.java

```diff
@@ -1,52 +1,67 @@
 package io.pinecone;
 
 import io.pinecone.model.CreateIndexRequest;
-import org.asynchttpclient.AsyncHttpClient;
-import org.asynchttpclient.DefaultAsyncHttpClient;
-
+import okhttp3.*;
 import java.io.IOException;
+import static io.pinecone.utils.Constants.*;
 
 public class PineconeIndexOperationClient {
-    private AsyncHttpClient client;
-    private PineconeClientConfig clientConfig;
-    private String url;
+    private final OkHttpClient client;
+    private final PineconeClientConfig clientConfig;
+    private final String url;
 
-    PineconeIndexOperationClient(PineconeClientConfig clientConfig, AsyncHttpClient client) {
+    PineconeIndexOperationClient(PineconeClientConfig clientConfig, OkHttpClient client) {
         this.clientConfig = clientConfig;
         this.client = client;
-        this.url = "https://controller." + clientConfig.getEnvironment() + ".pinecone.io/databases/";
+        this.url = BASE_URL_PREFIX + clientConfig.getEnvironment() + BASE_URL_SUFFIX;
     }
 
     public PineconeIndexOperationClient(PineconeClientConfig clientConfig) {
         this.clientConfig = clientConfig;
-        client = new DefaultAsyncHttpClient();
-        this.url = "https://controller." + clientConfig.getEnvironment() + ".pinecone.io/databases/";
+        this.client = new OkHttpClient();
+        this.url = BASE_URL_PREFIX + clientConfig.getEnvironment() + BASE_URL_SUFFIX;
     }
 
     public void deleteIndex(String indexName) throws IOException {
-        System.out.println("Sending delete index request:");
-        client.prepare("DELETE", url + indexName)
-                .setHeader("accept", "text/plain")
-                .setHeader("Api-Key", clientConfig.getApiKey())
-                .execute()
-                .toCompletableFuture()
-                .join();
-
-        client.close();
+        Request request = new Request.Builder()
```

Comment:
I'd a choice to chose between async vs sync calls. The older implementation used to use async calls but the join() was called right after the completableFuture() which I believe is similar to sync calls since it blocks and waits for the completion of the asynchronous request. So I decided to go with sync calls to keep things simpler.

Reply:
Makes sense, thanks for the explanation.